File: ClientApp/src/app/models/Post.ts
import { Reply } from "./Reply";
import { User } from "./User";

export interface Post {
    id?: number, 
    boardId: number,
    board?: string, 
    title: string, 
    user?: User, 
    userId: number,
    replies: number, 
    views: number, 
    body: string,
    createdAt?: Date, 
    replyMessages?: Reply[]
}

File: Models/Post.cs
using System.ComponentModel.DataAnnotations;

namespace ForumProject.Models
{
    public class Post
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Title { get; set; }
        [Required]
        public string Body { get; set; }
        [Required]
        public int UserId { get; set; }
        public User? User { get; set; }
        [Required]
        public DateTime CreatedAt { get; set; }
        [Required]
        public int Replies { get; set; }
        [Required]
        public int Views { get; set; }
        [Required]
        public int BoardId { get; set; }
        public Board? Board { get; set; }
        public IEnumerable<Reply>? ReplyMessages { get; set; }
    }
}